### PR TITLE
[MVT] Query mvt feature F-007

### PR DESCRIPTION
**Pull request by loicgasser**
<jenkins>A test link will be added here if the deploy on int succeeded.</jenkins>

@davidoesch you can already test with openmaptiles. That is what I understood from the specifications.

You can already make suggestions if you want.
image

<jenkins>[Test link](https://mf-geoadmin3.int.bgdi.ch/query_mvt_feature/index.html)</jenkins>

### CODE REVIEW

**Comment by davidoesch** [comment]
+1  well done. the key/values are the depending on the delivery by Karto
Question: multilang- needs to be tested

**Comment by oterral** [comment]
@davidoesch  Why not having a popup display an the map ?  I think it s really hard to understand and to find the functionnality as it is. I spend 5 minutes to find something with something displayed in the value field.
I think a popup like maputnik inspect mode is better.

**Comment by davidoesch** [comment]
@oterral  Agree, expected a little bit more info as well. The goal is to show the potential of vector attributes.
Thing is: background maps don't have tooltips, this is why I proposed  the footer info. at the same time clicking on maps is a PITA - the idea is to have a continous update of the info what is below the mouse pointer (which works only on desktop anyway)
Proposal: if you can add  maputnik style inspect mode using our tooltip CI/CD: great  (https://cloud.githubusercontent.com/assets/1288339/21468906/21cdb3d6-ca2e-11e6-955e-0e83b096fdba.png

**Comment by loicgasser** [comment]
I agree inspect mode would make more sense. I can easily adapt it. Should it be a toggle button in the footer?

File: src/components/queryvector/QueryVectorDirective.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+(function() {
+  var module = angular.module('ga_query_vector_directive', [
+    'ga_layerfilters_service'
+  ]);
+
+  var registerPointerMove = function(scope, map) {
+    return map.on('pointermove', function(evt) {
+      var featureFound;
+      map.forEachFeatureAtPixel(evt.pixel, function(feature, layer) {
```
[the remark this answers is not in the document]
Comment (loicgasser) [reply src/components/queryvector/QueryVectorDirective.js]: Thanks

**Comment by loicgasser** [comment]
@davidoesch I switched to an inspect mode version. Do you like it better?

**Comment by oterral** [comment]
small improvments:
 - seperate feature's result with css
 - alphabetical order of the properties

**Comment by loicgasser** [comment]
@oterral I ll create a table with the results and sort alphabetically. I ll try to make it a little seixier as well.
@davidoesh yes no problem, I ll add it in the feedback panel, it makes sense.

**Comment by loicgasser** [comment]
This is now a fork of https://github.com/geoadmin/mf-geoadmin3/pull/4563, merge this one first.

Fix for https://github.com/geoadmin/mf-geoadmin3/issues/4562
Inspect can now be activated in the feedback panel.
Keys will be translated when we have real data.

**Comment by loicgasser** [comment]
image

**Comment by davidoesch** [comment]
On mobile or does not fit the screen

IMHO: completely deactivate inspect on mobile. Remove it from panel if a touch device is activated

**Comment by loicgasser** [comment]
Good Point i forgot to remove it on mobile

**Comment by loicgasser** [comment]
also a fix for https://github.com/geoadmin/mf-geoadmin3/issues/4546

File: src/components/queryvector/style/queryvector.less [hunk 2]
```diff
@@ -0,0 +1,62 @@
+.ga-query-vector-popup {
+  position: absolute;
+  background-color: white;
+  -webkit-filter: drop-shadow(0 1px 4px rgba(0, 0, 0, 0.2));
```
Comment (oterral) [inline src/components/queryvector/style/queryvector.less]: no needed , autoprefixer will add it

Comment (loicgasser) [reply src/components/queryvector/style/queryvector.less]: Ok

**Comment by loicgasser** [comment]
* the popup stays open when you mouse is on another element (like the bg selector or outside the window. it should be closed.
* the popup keeps the information of the last feature if no feature found.

done and done, it does feel better, if no feature is found I hide the popup entirely. I could also display an empty popup, but it looks a bit weird.


* we should remove the non bg vt layer from the bg list in the feedback directive.

I deactivate all the vector features when a non VT layer is added to the map. I think this makes sense the way it is. removing it from the dropdown is easy though if @davidoesch prefers it.

**Comment by oterral** [comment]
> I deactivate all the vector features when a non VT layer is added to the map. I think this makes sense the way it is. removing it from the dropdown is easy though if @davidoesch prefers it.

I know  but for me it's not enough obvious what the user must do to have the style modification back

**Comment by oterral** [comment]
looks very good thx

**Comment by davidoesch** [comment]
"I deactivate all the vector features when a non VT layer is added to the map. I think this makes sense the way it is. removing it from the dropdown is easy "
Makes sense . Agree it is hard to know which layers have feedback but I propose that we start test.map.geo.admin.ch with a vt layer . So user can reset the viewer and can start over

**Pull request merged**